openquake/risklib/valid.py

```diff
@@ -835,6 +835,18 @@ def positiveints(value):
     return ints
 
 
+def weights(value):
+    """
+    >>> weights('0.4 0.3')
```

Comment:
The doctest is already self-explanatory. Anyway, it might be good to specify that `value` has to be a string of space-separated numbers.
We might also consider the possibility of accepting numbers such as '1 4', meaning that weights are in the proportion of 1 to 4. These would be transformed automatically, in order to make them sum up to 1, i.e. [0.2, 0.8]